infra/modules/kubevela-addons/definitions/nixbuildnet.cue

```diff
@@ -0,0 +1,94 @@
+parameter: {
+    secretName: string
```

Comment:
Ideally this wouldn't be needed, the secret would just be available everywhere (perhaps using the external-secrets thing)

Reply:
So can we assume the secret name will always be "nbn"? Do you suggest something else?
I can leave this parameter as optional with something as default value.

Reply:
@shlevy Implemented on https://github.com/input-output-hk/sc-dev-platform/pull/126/commits/ab43c58660c07dbbbaac20b1cbe60851a213f457

Reply:
We can't assume `nbn` unless we also make sure the secret is always available.

Reply:
hmmm thinking...

If we leave "_nbn_" as the default secret name and it's not available in the namespace, the container will likely fail with `CreateContainerConfigError` until the necessary secret becomes available. The same behavior will happen if the Application is setting parameter.secretName as "_somethingelse_" and the secret "_somethingelse_" is not available.

As we don't have a default mechanism to create/distribute secrets yet I am inclined to keep that behavior and add some notes in the documentation about the secret dependency. Once we have a default mechanism for the secrets management, we can update the trait to perform it automatically.

What do you think about ?

Reply:
Sounds good